### PR TITLE
Migrate erfinv and erfinv_ from the TH to Aten(CPU)

### PR DESCRIPTION
**Pull request by ifedan**
#24698

### CODE REVIEW

File: aten/src/ATen/native/cpu/UnaryOpsKernel.cpp [hunk 1]
```diff
@@ -258,6 +258,7 @@ IMPLEMENT_FLOAT_KERNEL(FLOATING, cos)
 // IMPLEMENT_FLOAT_KERNEL(FLOATING, cosh)
 IMPLEMENT_FLOAT_KERNEL(FLOATING, erf)
 IMPLEMENT_FLOAT_KERNEL(FLOATING, erfc)
+IMPLEMENT_FLOAT_KERNEL(FLOATING, erfinv)
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/cpu/UnaryOpsKernel.cpp]: I prefer if we clearly state that erfinv doesn't support AVX2 vectorization (dispatch it via `cpu_loop`). That might even make code faster, and definitely will generate smaller kernels.

**Review by VitalyFedyunin — CHANGES_REQUESTED** [review]
Please remove `TH_erfinv` and usage of it.

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -4063,16 +4056,23 @@
 - func: erfinv.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
   named_guard: False
   dispatch:
-    CPU: legacy::cpu::_th_erfinv_out
+    CPU: _erfinv_out_cpu
```
[the remark this answers is not in the document]
Comment (ifedan) [reply aten/src/ATen/native/native_functions.yaml]: I used IMPLEMENT_UNARY_OP_VEC macros, which generates these names.

File: aten/src/ATen/native/cpu/UnaryOpsKernel.cpp (hunk 1 above)
Comment (ifedan) [reply aten/src/ATen/native/cpu/UnaryOpsKernel.cpp]: > Please remove `TH_erfinv` and usage of it.

Done

**Comment by VitalyFedyunin** [comment]
Looks good, please attache benchmarking results.

**Comment by ifedan** [comment]
> Looks good, please attache benchmarking results.

image

**Comment by VitalyFedyunin** [comment]
> > Looks good, please attache benchmarking results.
> 
> image

single threaded?

**Comment by ifedan** [comment]
> > > Looks good, please attache benchmarking results.
> > 
> > 
> > image
> 
> single threaded?

yes

**Comment by ifedan** [comment]
**Single thread**image

**Comment by pbelevich** [comment]
Add erfinv to test/test_torch.py:test_unary_out_op_mem_overlap

**Review by VitalyFedyunin — APPROVED** [review]
Please address overlap tests in CUDA porting.

**Comment by VitalyFedyunin** [comment]
Please MOVE entire Copyright message.

**Pull request closed without merging**